src/trace.ml

```diff
@@ -83,6 +86,79 @@ module Make_commands (Backend : Backend_intf.S) = struct
     let filename ~record_dir = record_dir ^/ "hits.sexp"
   end
 
+  module Aaaa = struct
+    type t =
+      { push_traps : int64 list
+      ; pop_traps : int64 list
+      ; enter_traps : int64 list
+      }
+
+    let read_note cursor ~actual_base =
+      let descsz =
+        Owee_elf_notes.read_desc_size ~expected_owner:"OCaml" ~expected_type:1 cursor
+      in
+      if descsz < 8 * 4
+      then Owee_buf.invalid_format (Printf.sprintf "Too small size of note %d\n" descsz);
+      let recorded_base = Owee_buf.Read.u64 cursor in
+      let rec loop acc =
+        let addr = Owee_buf.Read.u64 cursor in
+        if Int64.equal addr 0L
+        then acc
+        else (
+          let addr = Owee_elf_notes.Stapsdt.adjust addr ~actual_base ~recorded_base in
+          loop (addr :: acc))
+      in
+      (* order of field initializers matters!! keep in sync with Emit.mlp *)
+      let enter_traps = loop [] in
+      let push_traps = loop [] in
+      let pop_traps = loop [] in
+      { enter_traps; push_traps; pop_traps }
+    ;;
+
+    let print msg l =
+      Core.print_string msg;
+      List.iter ~f:(fun a -> Core.printf " 0x%Lx" a) l;
+      Core.print_endline ""
+    ;;
+
+    let print_traps t =
+      print "enter_traps:" t.enter_traps;
+      print "push_traps:" t.push_traps;
+      print "pop_traps:" t.pop_traps
+    ;;
+
+    let make elf =
+      let path = Elf.filename elf in
+      Core.print_s [%message "file" (path : string)];
+      let buffer = Owee_buf.map_binary path in
```

Comment:
This is duplicating the work of `map_binary` and `read_elf` from `core/elf.ml`

Reply:
Yeah, this is terrible code that's not quite ready for review, was focusing on getting it working :)